### PR TITLE
Remove dns-prefix as a required field.

### PR DESCRIPTION
**Pull request by brendandburns**
Synthesize `--dns-prefix` if it's not provided.

@colemickens @lostintangent

### CODE REVIEW

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -265,7 +266,13 @@ def acs_create(resource_group_name, deployment_name, dns_name_prefix, name, ssh_
      if raw=true
     :raises: :class:`CloudError<msrestazure.azure_exceptions.CloudError>`
     """
+    if not dns_name_prefix:
+        # Use subscription id to provide uniqueness and prevent DNS name clashes
+        _, subscription_id, _ = Profile().get_login_credentials(subscription_id=None)
+        dns_name_prefix = '{}-{}'.format(name, subscription_id[0:6])
```
Comment (colemickens) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: I fear this doesn't protect against all name clashes. Consider these commands run in a single subscription:

```
az resource group create --name colemick-prod --location westus
az resource group create --name colemick-test --location westus
az acs create --orchestrator=kubernetes --resource-group=colemick-prod --name=kubecluster
az acs create --orchestrator=kubernetes --resource-group=colemick-test --name=kubecluster
```

Won't those result in conflicting dns prefixes, despite being a valid way of creating two clusters?

**Comment by colemickens** [comment]
Other comment, this change will create some inconsistency without doing this first or in tandem: [#1443: az acs kubernetes get-credentials should take resource_group/name rather than location/dns-prefix](https://github.com/Azure/azure-cli/issues/1443)

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py [hunk 2]
```diff
@@ -54,7 +54,7 @@ def _get_default_install_location(exe_name):
 register_cli_argument('acs', 'orchestrator_type', **enum_choice_list(ContainerServiceOchestratorTypes))
 #some admin names are prohibited in acs, such as root, admin, etc. Because we have no control on the orchestrators, so default to a safe name.
 register_cli_argument('acs', 'admin_username', options_list=('--admin-username',), default='azureuser', required=False)
-register_cli_argument('acs', 'dns_name_prefix', options_list=('--dns-prefix', '-d'))
+register_cli_argument('acs', 'dns_name_prefix', options_list=('--dns-prefix', '-d'), default=None, required=False)
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py]: Please remove `default=None, required=False` as there's no need to specify it here.
Instead, in the method definition for `acs_create()`, make `dns_name_prefix` have a default of `None` i.e. `dns_name_prefix=None`.
The CLI will automatically pick up that it's an optional parameter that way.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py (hunk 1 above)
Comment (brendandburns) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: added resource group.

**Comment by brendandburns** [comment]
Comments addressed, please re-check.

Thanks!
--brendan

**Pull request merged**